### ISSUE TITLE
Update Context._parse()?

### ISSUE THREAD

**Issue by jhpedemonte** [maqetta/maqetta#2702]
The `if (header.modules)` check in `Context.setHeader()` never seems to be true.

`header.modules` comes from `data.modules`, which is set in `Context._parse()`.  That whole function is pretty old and probably needs to be updated/rewritten -- especially this `modules` bit, which might not even be necessary any more.

**Comment by peller** [comment]
-> M8

**Comment by peller** [comment]
-> M9

**Comment by peller** [comment]
Indeed, header.modules seems to always be empty (not falsey, but an empty array)  The code in Context._parse that does

```
                // grab AMD-style dependencies
                text.replace(/require\(\[["']([^'"]+)["']\]\)/g, function(match, module) {
                    data.modules.push(module);
                });
```

doesn't match anything as it seems to be looking only for require statements with one argument on one line.  Probably something we did in the transition to AMD.  There's a similar check just above for dojo.require.  We probably can't remove this unless we're willing to drop support for old pre-AMD pages... if that even still works.  I don't think we include that in our tests.

**Comment by peller** [comment]
@JonFerraiolo @billreed63 It seems like the old code which parses the require/dojo.require statement no longer runs, and instead we do our loadRequire based on what's in data-dojo-type attributes.  Does that make sense?  In which case, I think Javier's original comment stands.
